app/src/main/java/com/hamdan/acceleromatorminiproject/view/MainActivity.kt

```diff
@@ -31,23 +38,44 @@ class MainActivity : ComponentActivity() {
         super.onCreate(savedInstanceState)
         val intent = Intent(this, TimerService::class.java)
         setContent {
-            MainPageContent(
-                state = sensorViewModel.state.value,
-                onReadyClicked = {
-                    if (isBound) {
+            val navController = rememberNavController()
+            NavHost(navController = navController, startDestination = MAIN_PAGE) {
+                composable(MAIN_PAGE) {
+                    MainPageContent(
+                        state = sensorViewModel.state.value,
+                        onReadyClicked = {
+                            if (isBound) {
+                                unbindService()
+                                bindService()
+                            } else {
+                                bindService()
```

Comment:
You're doing this in either branch

Reply:
I didn't adjust this if statement, only moved it

Regardless, I have adjusted to simply bind the service